robocin/geometry/README.md

```diff
@@ -0,0 +1,185 @@
+
+### Static Constructors
+
+- `static Point2D origin()`: Returns a point representing the origin (0, 0).
+- `static Point2D fromPolar(value_type angle) requires(std::floating_point<value_type>)`: Creates a point from polar
+  coordinates with the given angle.
+- `static Point2D fromPolar(value_type angle, value_type length) requires(std::floating_point<value_type>)`: Creates a
+  point from polar coordinates with the given angle and length.
+
+### Validators
+
+- `bool isNull() const`: Checks if the point is null (coordinates are both zero). Returns `true`
+  if the point is null, `false` otherwise.
+
+### Arithmetic-Assignment Operators
+
+- `Point2D& operator+=(const Point2D& other)`: Adds the coordinates of another point to this point and updates its
+  value. Returns a reference to the modified point.
+- `Point2D& operator-=(const Point2D& other)`: Subtracts the coordinates of another point from this point and updates
+  its value. Returns a reference to the modified point.
+- `Point2D& operator*=(value_type factor)`: Multiplies the coordinates of this point by a scalar factor and updates its
+  value. Returns a reference to the modified point.
+- `Point2D& operator/=(value_type factor)`: Divides the coordinates of this point by a scalar factor and updates its
+  value. Returns a reference to the modified point.
+
+### Arithmetic Operators
+
+- `Point2D operator+(const Point2D& other) const`: Addition operator that returns the sum of two points.
+- `Point2D operator-(const Point2D& other) const`: Subtraction operator that returns the difference between two points.
+- `Point2D operator*(value_type factor) const`: Multiplication operator that scales the point by a factor.
+- `Point2D operator/(value_type factor) const`: Division operator that scales the point by the inverse of a factor.
+
+### Arithmetic Friend Operators
+
+- `friend Point2D operator*(value_type factor, const Point2D& point)`: Friend operator that allows multiplication of a
+  point by a factor in the reverse order.
+
+### Sign Operators
+
+- `Point2D operator+() const`: Unary plus operator that returns the point itself.
+- `Point2D operator-() const`: Unary minus operator that returns the negation of the point.
+
+### Comparison Operators
+
+- `bool operator==(const Point2D& other) const`: Equality operator that checks if two points are equal.
+- `auto operator<=>(const Point2D& other) const`: Three-way comparison operator that compares two points and returns
+  their relative ordering.
+
+### Swap
+
+- `void swap(Point2D& other) noexcept`: Swaps the contents of the current point with another point.
+
+### Geometry
+
+- `value_type dot(const Point2D& other) const`: Computes the dot product of the current point and another point.
+- `value_type cross(const Point2D& other) const`: Computes the cross product of the current point and another point.
+- `value_type manhattanLength() const`: Computes the Manhattan distance between the origin and the current point.
```

Comment:
This a example of point only operation.

Reply:
I dont like that. So for manhattan that commonly a binary function, I have to do:
`(b-a).manhattanLength()`
Instead of:
`a.manhattanDistance(b)`?
I think that manhattan is more related to distance than length.

Reply:
Very well observed! Again, this was partly inspired by [`QPointF::manhattanLength`](https://doc.qt.io/qt-6/qpointf.html#manhattanLength).

Perhaps keeping both is reasonable?

Reply:
Yes.